feat(autoware_lanelet2_utils): replace ported functions from autoware_lanelet2_extension

sarun-hub opened the pull request:
## Description
Several functions from `autoware_lanelet2_extension` are ported to `autoware_lanelet_utils`. Those functions will be deprecated in `autoware_lanelet2_extension` ([PR #81](https://github.com/autowarefoundation/autoware_lanelet2_extension/pull/81)).
This PR will replace all the usages of ported function in autoware_universe.
Target functions
- [x] `getClosestCenterPose` - autoware_core [PR #693](https://github.com/autowarefoundation/autoware_core/pull/693).
- [x] `getClosestLaneletWithConstrains` - (include simulator component)
- [x] `getLaneletLength2d` - replaced by `lanelet::geometry::getlength2d`
- [x] `getLaneletLength3d` - replaced by `lanelet::geometry::getlength3d` (including perception component)
- [ ] <s>`lineStringWithWidthToPolygon`</s>
- [ ] <s>`lineStringToPolygon`</s>
- [x] `getConflictingLanelets` - autoware_core [PR #704](https://github.com/autowarefoundation/autoware_core/pull/704) (include control component)
- `getClosestSegment` - Leftover from [PR #11387](https://github.com/autowarefoundation/autoware_universe/pull/11387)

## Related links

**Parent Issue:**

- Link

<!-- ⬇️🟢
**Private Links:**

- [CompanyName internal link]()
⬆️🟢 -->

## How was this PR tested?

- [Common](https://evaluation.tier4.jp/evaluation/reports/e3fa5e4d-9abd-59bc-824c-e8a81ae47396?project_id=prd_jt)
- [DLR](https://evaluation.tier4.jp/evaluation/reports/fffc4f65-abe3-5a94-9992-cc276f9891f7?project_id=prd_jt)
- [FuncVerification](https://evaluation.tier4.jp/evaluation/reports/89dc6a03-9af4-5685-b1d4-d68f701c4fa2?project_id=prd_jt)

## Notes for reviewers

None.

## Interface changes

None.

<!-- ⬇️🔴

### Topic changes

#### Additions and removals

| Change type   | Topic Type      | Topic Name    | Message Type        | Description       |
|:--------------|:----------------|:--------------|:--------------------|:------------------|
| Added/Removed | Pub/Sub/Srv/Cli | `/topic_name` | `std_msgs/String`   | Topic description |

#### Modifications

| Version | Topic Type      | Topic Name        | Message Type        | Description       |
|:--------|:----------------|:------------------|:--------------------|:------------------|
| Old     | Pub/Sub/Srv/Cli | `/old_topic_name` | `sensor_msgs/Image` | Topic description |
| New     | Pub/Sub/Srv/Cli | `/new_topic_name` | `sensor_msgs/Image` | Topic description |

### ROS Parameter Changes

#### Additions and removals

| Change type   | Parameter Name | Type     | Default Value | Description       |
|:--------------|:---------------|:---------|:--------------|:------------------|
| Added/Removed | `param_name`   | `double` | `1.0`         | Param description |

#### Modifications

| Version | Parameter Name   | Type     | Default Value | Description       |
|:--------|:-----------------|:---------|:--------------|:------------------|
| Old     | `old_param_name` | `double` | `1.0`         | Param description |
| New     | `new_param_name` | `double` | `1.0`         | Param description |

🔴⬆️ -->

## Effects on system behavior

None.

ktro2828 reviewed (approved):
LGTM for `autoware_map_based_prediction` part.

sarun-hub commented:
@ktro2828 Sorry for additional fix :pray:.
There is leftover `getClosestSegment` so I removed it, and removed unused `lanelet2_extension` include. 
Please recheck it!

ktro2828 commented on perception/autoware_map_based_prediction/src/utils.cpp:
```diff
@@ -511,7 +511,8 @@ double lateral_distance_to_lanelet_bounds(
   auto distance = std::numeric_limits<double>::max();
   for (const auto & bound : {ll.leftBound(), ll.rightBound()}) {
     const auto p = lanelet::BasicPoint2d(point.x, point.y);
```
[IMO] Generating `p` can be moved to the line just before it needs because this might be unused if some conditions are not satisfied, or we can move it to the out of the loop because this is always created from the same `point`.

kyoichi-sugahara reviewed (approved):
LGTM for start_planner and planning_validator
